File: DSA/Arshdeep-Kaur/Day 30/Day30_Question2.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        int maxi = 0;
        int ans = 0;
        for (int i=0; i<nums.size(); i++) {
            if (nums[i] > maxi) {
                maxi = nums[i];
                ans = i;
            }
        }

        for(int i=0; i<nums.size(); i++) {
            if (nums[i] == maxi) {
                continue;
            }
            else {
                if (maxi >= 2*nums[i]) {
                    continue;
                }
                else {
                    return -1;
                }
            }
        }

        return ans;
    }
};

File: DSA/Pragya_Agarwal/Day30/Day30_Question2.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        max_val = nums[0]
        index = 0
        
        for i in range(1, len(nums)):
            if nums[i] > max_val:
                max_val = nums[i]
                index = i
        for i in range(len(nums)):
            if i != index and max_val < 2 * nums[i]:
                return -1
        return index
